src/mono/sample/iOS/Program.csproj

```diff
@@ -3,141 +3,57 @@
     <OutputType>Exe</OutputType>
     <OutputPath>bin</OutputPath>
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
-    <TargetOS Condition="'$(TargetOS)' == ''">ios</TargetOS>
-    <TargetOS Condition="'$(TargetsiOSSimulator)' == 'true'">iossimulator</TargetOS>
-    <DeployAndRun Condition="'$(DeployAndRun)' == ''">true</DeployAndRun>
-    <iOSLikeDedup Condition="'$(TargetOS)' == 'ios'">true</iOSLikeDedup>
+    <TargetOS Condition="'$(TargetOS)' == ''">iossimulator</TargetOS>
     <RuntimeIdentifier>$(TargetOS)-$(TargetArchitecture)</RuntimeIdentifier>
     <DefineConstants Condition="'$(ArchiveTests)' == 'true'">$(DefineConstants);CI_TEST</DefineConstants>
     <AppName>HelloiOS</AppName>
+    <MainLibraryFileName>Program.dll</MainLibraryFileName>
+    <DevTeamProvisioning Condition="'$(DevTeamProvisioning)' == ''">-</DevTeamProvisioning>
+    <RunAOTCompilation Condition="'$(RunAOTCompilation)' == ''">true</RunAOTCompilation>
+    <DeployAndRun Condition="'$(DeployAndRun)' == ''">false</DeployAndRun>
     <StripDebugSymbols Condition="'$(StripDebugSymbols)' == ''">false</StripDebugSymbols>
-  </PropertyGroup>
+    <UseMonoJustInterp Condition="'$(UseMonoJustInterp)' == '' and '$(MonoForceInterpreter)' == 'true'">true</UseMonoJustInterp>
 
-  <PropertyGroup>
     <PublishTrimmed>true</PublishTrimmed>
     <TrimMode>Link</TrimMode>
+    <Optimized Condition="'$(Configuration)' == 'Release'">True</Optimized>
+
+    <ShouldILStrip Condition="'$(RunAOTCompilation)' == 'true' and '$(UseMonoJustInterp)' != 'true'" >true</ShouldILStrip>
+    <EnableDefaultAssembliesToBundle>true</EnableDefaultAssembliesToBundle>
+    <AppleGenerateAppBundle>true</AppleGenerateAppBundle>
+    <GenerateXcodeProject>true</GenerateXcodeProject>
+    <UseConsoleUITemplate>false</UseConsoleUITemplate>
+
+    <BuildAppBundleAfterTargets>CopyFilesToPublishDirectory</BuildAppBundleAfterTargets>
+    <BuildAppBundleAfterTargets Condition="'$(ArchiveTests)' == 'true'">CopyFilesToPublishDirectory;Build</BuildAppBundleAfterTargets>
+    <BuildAppBundleDependsOnTargets Condition="'$(ArchiveTests)' == 'true'">Publish</BuildAppBundleDependsOnTargets>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TargetOS)' == 'maccatalyst'">
-    <DevTeamProvisioning Condition="'$(TargetOS)' == 'maccatalyst' and '$(DevTeamProvisioning)' == ''">adhoc</DevTeamProvisioning>
+    <DevTeamProvisioning Condition="'$(DevTeamProvisioning)' == ''">adhoc</DevTeamProvisioning>
```

Comment:
Great job on simplifying the sample project, it looks much better now!

However, I was wondering whether we could simplify these properties further and/or move them into to `AppleBuild.props`?
For example, setting `DevTeamProvisioning` on line 33 seems to be unreachable in case when `'$(TargetOS)' == 'maccatalyst'`, as it will be previously set to `-` on line 11.

Reply:
Thanks, good catch :) Some of them, like EnableDefaultAssembliesToBundle, AppleGenerateAppBundle, GenerateXcodeProject, and UseConsoleUITemplate are set in `tests.ioslike.targets` and used in `AppleBuild.targets`.

Once we enable the functional tests on a device, we may determine that some of them are redundant. I've excluded properties that set default value to false. Do you have other suggestions?